go/pkg/bertyprotocol/store_metadata_index.go

```diff
@@ -167,12 +168,39 @@ func (m *metadataStoreIndex) handleGroupAddDeviceSecret(event proto.Message) err
 	}
 
 	if m.ownMemberDevice.device.Equals(senderPK) {
-		m.sentSecrets[string(e.DestMemberPK)] = struct{}{}
+		m.sentSecrets[string(e.DevicePK)+string(e.DestMemberPK)] = struct{}{}
+		if chanSentSecrets, ok := m.chanSentSecrets[string(e.DevicePK)+string(e.DestMemberPK)]; ok {
+			chanSentSecrets <- struct{}{}
+		}
 	}
 
 	return nil
 }
 
+func (m *metadataStoreIndex) watchSentSecrets(memberPK crypto.PubKey) (chan struct{}, error) {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+
+	memberRaw, err := memberPK.Raw()
+	if err != nil {
+		return nil, errcode.ErrDeserialization.Wrap(err)
+	}
+	ownDeviceRaw, err := m.ownMemberDevice.device.Raw()
+	if err != nil {
+		return nil, errcode.ErrDeserialization.Wrap(err)
+	}
+
+	m.logger.Debug("watchSentSecrets", zap.Any("ownDevice", ownDeviceRaw), zap.Any("memberPK", memberRaw))
+
+	channel, ok := m.chanSentSecrets[string(ownDeviceRaw)+string(memberRaw)]
```

Comment:
```suggestion
 	mappingKey := fmt.Sprintf("%s%s", ownDeviceRaw, memberRaw)
	channel, ok := m.chanSentSecrets[mappingKey]
```

reusable few lines after

Reply:
Resolved because this code will be removed, see #2312 (comment)